sdk/monitor/azure-monitor-opentelemetry-exporter/src/main/java/com/azure/monitor/opentelemetry/exporter/policy/AzureMonitorRedirectCustomPolicy.java

```diff
@@ -0,0 +1,92 @@
+     */
+    public AzureMonitorRedirectCustomPolicy() {
+        maxRedirectRetries = 10;
+    }
+
+    /**
+     * Creates {@link AzureMonitorRedirectCustomPolicy} with provided maxRedirectRetries.
+     *
+     * @param maxRedirectRetries The maximum retries through pipeline when a redirect is received as response
+     */
+    public AzureMonitorRedirectCustomPolicy(int maxRedirectRetries) {
+        this.maxRedirectRetries = maxRedirectRetries;
+    }
+
+    @Override
+    public Mono<HttpResponse> process(HttpPipelineCallContext context, HttpPipelineNextPolicy next) {
+        return attemptRetry(context, next, context.getHttpRequest(), 0);
+    }
+
+    /**
+     *  Function to process through the HTTP Response received in the pipeline
+     *  and retry sending the request with new redirect url.
+     */
+    private Mono<HttpResponse> attemptRetry(final HttpPipelineCallContext context,
+                                            final HttpPipelineNextPolicy next,
+                                            final HttpRequest originalHttpRequest,
+                                            final int retryCount) {
+        // make sure the context is not modified during retry, except for the URL
+        context.setHttpRequest(originalHttpRequest.copy());
+        if (this.redirectedEndpointUrl != null) {
+            context.getHttpRequest().setUrl(this.redirectedEndpointUrl);
+        }
+        return next.clone().process()
+            .flatMap(httpResponse -> {
+                if (shouldRetryWithRedirect(httpResponse.getStatusCode(), retryCount)) {
+                    String responseLocation = httpResponse.getHeaderValue("Location");
+                    if (responseLocation != null) {
+                        this.redirectedEndpointUrl = responseLocation;
+                        return attemptRetry(context, next, originalHttpRequest, retryCount + 1);
+                    }
+                }
+                return Mono.just(httpResponse);
+            }).onErrorResume(err -> Mono.error(new RuntimeException(
+                String.format("Retry with redirect error details: %s", err.getMessage()),
+                err)));
+    }
+
+    /**
+     * Determines the delay duration that should be waited before retrying.
+     *
+     * @param statusCode HTTP response status code
+     * @param tryCount Redirect retries so far
+     * @return True if statusCode corresponds to HTTP redirect response codes and redirect
+     * retries is less than {@code maxRedirectRetries}.
+     */
+    private boolean shouldRetryWithRedirect(int statusCode, int tryCount) {
+        if (tryCount >= maxRedirectRetries) {
+            logger.verbose("Max redirect retries limit reached:%d.", maxRedirectRetries);
+            return false;
+        }
+        return statusCode == HttpURLConnection.HTTP_MOVED_TEMP
+                || statusCode == HttpURLConnection.HTTP_MOVED_PERM
+                || statusCode == 308;
```

Comment:
Create a constant for this status code in this class if it is not available in HttpURLConnection.